### PR TITLE
Fixing runtimes

### PR DESCRIPTION
**Pull request by Sindorman**
This PR tracks down and fixes runtimes caught in logs on live. Summary:

- Adds Travis master branch build status into the README

- Sanity check when Astar returns null as list

- Sanity check for like 519 in `mob_helpers.dm` because of runtime `Cannot read null.z`

- Rings were runtiming because they were missing few var for carrying blood for forensic purposes

- Simple mobs and any other mobs no longer runtime on electrocuted_act. Simple mobs now get's sparks and sound when electrocuted.

- Hostile mobs now attack things that stand in their way to their target - windows and grills. 

- Hostile mobs now do custom emotes when they find their target that is a living being.

- Fix for lightning_source.dm that occurs when our corners are null. Provided by @Lohikar

### CODE REVIEW

File: code/modules/clothing/clothing.dm [hunk 1]
```diff
@@ -366,6 +366,9 @@
 		"Resomi" = 'icons/mob/species/resomi/gloves.dmi'
 		)
 
+	var/transfer_blood = 0
+	var/mob/living/bloody_hands_mob = null
```
Comment (skull132) [inline code/modules/clothing/clothing.dm]: Hard ref to mob is a no go. Use a weakref datum.

File: code/modules/clothing/clothing.dm [hunk 2]
```diff
@@ -971,3 +976,6 @@
 	slot_flags = SLOT_GLOVES
 	gender = NEUTER
 	var/undergloves = 1
+
+	var/transfer_blood = 0
+	var/mob/living/bloody_hands_mob = null
```
Comment (skull132) [inline code/modules/clothing/clothing.dm]: Hard ref to mob is a no go. Use a weakref datum.

**Comment by skull132** [comment]
My review stands, just as an FYI. If a bug fix must refactor code, then so be it.

**Comment by Sindorman** [comment]
That's fine, I am working on it. So going WIP for a bit

**Comment by Sindorman** [comment]
@skull132 I have take out commits for rings and made a separate PR for it, please review and merge this one in the meantime.

File: code/modules/mob/living/bot/secbot.dm [hunk 3]
```diff
@@ -361,6 +361,7 @@
 /mob/living/bot/secbot/proc/calc_path(var/turf/avoid = null)
 	path = AStar(loc, patrol_target, /turf/proc/CardinalTurfsWithAccess, /turf/proc/Distance, 0, 120, id=botcard, exclude=avoid)
 	if(!path.len)
+		path = list()
```
Comment (skull132) [inline code/modules/mob/living/bot/secbot.dm]: This doesn't make sense. If accessing `path.len` doesn't runtime, and the list has no length, why are we swapping it for a new list _now_? The effects of `path = list()` are already a requirement for us to get here.

**Pull request merged**